### ISSUE TITLE
builder should not assume all modules starting with "@" are system

### ISSUE THREAD

**Issue by flyingmutant** [systemjs/builder#340]
[This](https://github.com/systemjs/builder/blob/master/lib/trace.js#L66-L68) does not work with e.g. https://github.com/ReactiveX/RxJS, which is installed with `npm install @reactivex/rxjs`.

Angular 2 uses `require('@reactivex/rxjs/…')`, so builder can not bundle current Angular 2 because of this.

**Comment by guybedford** [comment]
Thanks for reporting. The workaround here would be to use a paths configuration to alter the canonical naming:

```javascript
System.config({
  paths: { 
    'reactivex/*': '@reactivex/rxjs/*'
  }
});
```

That should make bundling work without any other changes being necessary.

Then in terms of the underlying cause, we do need a comprehensive way to tell the difference between modules that were "set" as system modules in the registry and modules which are bundleable. Simply checking that a given module never went through the loader hooks would provide this information, so it shouldn't be a problem to implement.

**Comment by flyingmutant** [comment]
Thanks for the quick response! `paths` workaround does work indeed (together with `map` to map to `node_modules` subdirectory, and `packages` entry with `defaultExtension: js`).

**Comment by guybedford** [comment]
Released in 0.14.8.

**Comment by flyingmutant** [comment]
Thanks for fixing — can confirm everything is good now.
